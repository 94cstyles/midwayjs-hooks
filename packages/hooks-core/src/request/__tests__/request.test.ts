import { request } from '..'
import { mocked } from 'ts-jest/utils'
import axios from 'axios'

jest.mock('axios')

const mockedAxios = mocked(axios)

test('should call axios', () => {
  request({
    url: '/',
    method: 'GET',
    data: { args: [] },
    meta: {},
  })
  request({
    url: '/',
    method: 'GET',
    data: { args: [] },
    meta: {},
  })
  request({
    url: '/api/books',
    method: 'GET',
    data: { args: [] },
    meta: {},
  })

  expect(mockedAxios.mock.calls).toHaveLength(3)
  expect(mockedAxios.mock.calls).toMatchSnapshot()
})

afterAll(() => {
  jest.unmock('axios')
})
