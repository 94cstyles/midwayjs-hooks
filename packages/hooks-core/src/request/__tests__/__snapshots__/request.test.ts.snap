// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should call axios 1`] = `
Array [
  Array [
    Object {
      "data": Object {
        "args": Array [],
      },
      "meta": Object {},
      "method": "GET",
      "url": "/get",
    },
  ],
  Array [
    Object {
      "data": Object {
        "args": Array [
          1,
          2,
          3,
        ],
      },
      "meta": Object {},
      "method": "POST",
      "url": "/post",
    },
  ],
  Array [
    Object {
      "data": Object {
        "args": Array [],
      },
      "meta": Object {},
      "method": "GET",
      "url": "/",
    },
  ],
  Array [
    Object {
      "data": Object {
        "args": Array [],
      },
      "meta": Object {},
      "method": "GET",
      "url": "/api/books",
    },
  ],
]
`;
